lib/services/elasticsearch.js

```diff
@@ -1003,7 +1003,11 @@ class ElasticSearch extends Service {
 
     try {
       if (!_.isEmpty(mappings)) {
+        const previousMappings = await this.getMapping(index, collection);
         await this.updateMapping(index, collection, mappings);
+        if (_.isMatch(previousMappings, { dynamic: 'false'})) {
```

Comment:
You should walk the entire previous mapping content to search for former `dynamic: false` property since it could have been set on any level.  
Example:
```
{
  dynamic: true,
  properties: {
    name: { type: 'keyword' },
    metadata: {
      dynamic: false,
      properties: {  }
    }
  }
}
```

Reply:
dang, I read the lodash documentation wrong, I saw the word "deep" and I assumed it made a deep seach... turns out, it "only" performs a deep comparison, which is a bit different :sweat_smile: 

Good catch :+1:

Reply:
:disappointed: In deed, I missed it too